src/export/StructureDefinitionExporter.ts

```diff
@@ -420,6 +411,124 @@ export class StructureDefinitionExporter implements Fishable {
+          let contextItem = splitContext[0];
+          let contextPath = splitContext.slice(1).join('#');
+          if (contextPath === '' && !isUri(contextItem)) {
+            const splitPath = splitOnPathPeriods(contextItem);
+            contextItem = splitPath[0];
+            contextPath = splitPath.slice(1).join('.');
+          }
+          const targetResource = this.fishForFHIR(
+            contextItem,
+            Type.Extension,
+            Type.Profile,
+            Type.Resource,
+            Type.Logical,
+            Type.Type
+          );
+          if (targetResource != null) {
+            const resourceSD = StructureDefinition.fromJSON(targetResource);
+            const targetElement = resourceSD.findElementByPath(contextPath, this);
+            if (targetElement != null) {
+              // we want to represent the context using "extension" type whenever possible.
+              // if the resource is an Extension, and every element along the path to the target
+              // has type "extension", then this context can be represented with type "extension".
+              // otherwise, this is "element" context.
+              if (targetElement.isPartOfComplexExtension()) {
+                this.setContextForComplexExtension(structDef, targetElement, extContext);
+              } else {
+                let contextExpression: string;
+                let contextType = 'element';
+                if (resourceSD.type === 'Extension' && targetElement.parent() == null) {
+                  contextExpression = resourceSD.url;
+                  contextType = 'extension';
+                } else if (
+                  resourceSD.derivation === 'specialization' &&
+                  resourceSD.url.startsWith('http://hl7.org/fhir/StructureDefinition/')
+                ) {
+                  contextExpression = targetElement.id;
+                } else {
+                  contextExpression = `${resourceSD.url}#${targetElement.id}`;
+                }
+                structDef.context.push({
+                  expression: contextExpression,
+                  type: contextType
+                });
+              }
+            } else {
+              logger.error(
+                `Could not find element ${contextPath} on resource ${contextItem} to use as extension context.`,
+                extContext.sourceInfo
+              );
+            }
+          } else {
+            logger.error(
+              `Could not find resource ${contextItem} to use as extension context.`,
+              extContext.sourceInfo
+            );
+          }
+        }
+      });
+    } else {
+      structDef.context = [
+        {
+          type: 'element',
+          expression: 'Element'
+        }
+      ];
+    }
+  }
+
+  private setContextForComplexExtension(
+    structDef: StructureDefinition,
+    targetElement: ElementDefinition,
+    extContext: ExtensionContext
+  ) {
+    const extensionHierarchy = [
```

Comment:
I think I'd appreciate a comment here indicating the target format you're going for and maybe even including a direct link to Lloyds message where he says this is what to do.